### DISCUSSION TITLE
Dynamically Change Dataset and DataLoader

### DISCUSSION THREAD

**Discussion by DeshanKTD** [MIC-DKFZ/nnUNet#2841] [Q&A]
Is there a way to dynamically select dataset and dataloader as we do in trainers?

**Comment by Rainbow96r** [comment]
try this,bro
class nnUNetTrainer_CustomDataLoader(nnUNetTrainer):
    def get_plain_dataloaders(self, initial_patch_size: Tuple[int, ...], dim: int):
        # Select different data loaders based on conditions
        if self.configuration_manager.num_dimensions == 2:
            if hasattr(self, 'use_balanced_sampling') and self.use_balanced_sampling:
                dataloader_tr = BalancedSamplingDataLoader2D(
                    data=self.dataset_tr,
                    batch_size=self.configuration_manager.batch_size,
                    patch_size=initial_patch_size,
                    final_patch_size=initial_patch_size,  # Perform all transformations in the data loader
                    oversample_foreground_percent=self.oversample_foreground_percent
                )
            else:
                dataloader_tr = nnUNetDataLoader2D(
                    data=self.dataset_tr,
                    batch_size=self.configuration_manager.batch_size,
                    patch_size=initial_patch_size,
                    final_patch_size=initial_patch_size,
                    oversample_foreground_percent=self.oversample_foreground_percent
                )
        else:
            # Similar for 3D case
            dataloader_tr = nnUNetDataLoader3D(...)
        
        dataloader_val = nnUNetDataLoader2D(...)  # Validation set usually doesn't require special handling
        
        return dataloader_tr, dataloader_val
